kernel/arch/x86_64/Rules.mk

```
@@ -9,8 +7,10 @@
 ##
 ################################################################################
 
-set -eu
 
-mkdir -p "${BUILD_DIR}/boot"
-nasm -f elf64 "kernel/arch/${ARCH}/boot/multiboot2_header.asm" -o "${BUILD_DIR}/boot/multiboot2_header.o"
-nasm -f elf64 "kernel/arch/${ARCH}/boot/boot.asm" -o "${BUILD_DIR}/boot/boot.o"
+ARCH_SRCS := $(wildcard kernel/arch/$(ARCH)/boot/*.asm)
+ARCH_OBJECTS := $(addprefix $(TARGET_DIRECTORY)/, $(ARCH_SRCS:kernel/arch/$(ARCH)/%.asm=%.o))
+
+$(TARGET_DIRECTORY)/boot/%.o:	kernel/arch/$(ARCH)/boot/%.asm
```

```suggestion
$(TARGET_DIRECTORY)/%.o:	kernel/arch/$(ARCH)/%.asm
```

(Please test this, just in case I messed up something)